G1ANT.Addon.MSOffice/Commands/ExcelOpenCommand.cs

```diff
@@ -37,19 +35,20 @@ public ExcelOpenCommand(AbstractScripter scripter) : base(scripter)
         {
         }
 
-
         public void Execute(Arguments arguments)
         {
             try
             {
                 ExcelWrapper excelWrapper = ExcelManager.CreateInstance();
                 excelWrapper.Open(arguments.Path?.Value, arguments.Sheet?.Value, !arguments.InBackground.Value);
-                Scripter.Variables.SetVariableValue(arguments.Result.Value, new Language.IntegerStructure(excelWrapper.Id));
+                Scripter.Variables.SetVariableValue(arguments.Result.Value, new IntegerStructure(excelWrapper.Id));
             }
             catch (Exception ex)
             {
-                throw new ApplicationException($"Problem occured while opening excel instance. Path: '{arguments.Path.Value}', Sheet: '{arguments.Sheet?.Value}', InBackground: '{arguments.InBackground.Value}'", ex);
+                if (ex.GetType() == typeof(COMException) && ex.Message.Contains("80040154"))
```

Comment:
catch (COMException ex)
{
    if (ex.Message.Contains("80040154"))
        throw new Exception("Could not find Microsoft Office on computer. Please make sure it is installed and try again.");
    throw;
}
catch (exception ex)
{
    throw new ApplicationException($"Problem occured while opening excel instance. Path: '{arguments.Path?.Value}', Sheet: '{arguments.Sheet?.Value}', InBackground: '{arguments.InBackground.Value}'", ex);
}
}